parquet-column/src/main/java/org/apache/parquet/schema/Type.java

```diff
@@ -314,4 +314,12 @@ void checkContains(Type subType) {
    */
    abstract <T> T convert(List<GroupType> path, TypeConverter<T> converter);
 
+  /**
+   * Returns the {@link Type} specific comparator for properly comparing values. The natural ordering of the values
+   * might not proper in certain cases (e.g. {@code UINT_32} requires unsigned comparison of {@code int} values while
+   * the natural ordering is signed.)
+   */
+  public <T> PrimitiveComparator<T> comparator() {
```

Comment:
I don't think it is necessary to expose this on `Type`. It is fine to ensure the type is a primitive by calling `type.asPrimitive().comparator()`.